### PR TITLE
Update track api

### PR DESCRIPTION
**Pull request by kennyklaviyo**
# Purpose

**Type of PR**
* [ ] Bug Fix
* [ ] Feature Work
* [ ] Revert
* [ ] Refactor / Code Cleanup
* [ ] Other (fill in...)

### CODE REVIEW

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/KlaviyoRequest.kt [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
-     * Builds out our query data and sends the request to Klaviyo
+     * Encodes the given string into a non-wrapping base64 string
+     * Needed to parse the query data into encoded information
+     *
+     * @param data the string we want to encode
      */
-    override fun sendNetworkRequest(): String? {
-        queryData = buildKlaviyoJsonQuery()
-        return super.sendNetworkRequest()
+    internal fun encodeToBase64(data: String): String {
```
Comment (ndurell) [inline sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/KlaviyoRequest.kt:31]: Do you still need this?

Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/KlaviyoRequest.kt:31]: Oh maybe you are using get still ok...

Comment (kennyklaviyo) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/KlaviyoRequest.kt:31]: the identify still uses it

File: sdk/build.gradle [hunk 2]
```diff
@@ -46,14 +47,27 @@ subprojects {
...
         androidTestImplementation "androidx.test:runner:$ext.testRunnerVersion"
         androidTestImplementation "androidx.test.espresso:espresso-core:$ext.espressoVersion"
     }
+
+    ktlint {
+        verbose = true
+        android = true
+        ignoreFailures = false
+        outputToConsole = true
+        outputColorName = "RED"
```
Comment (ndurell) [inline sdk/build.gradle:65]: Maybe maroon instead...less harsh.

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt [hunk 3]
```diff
@@ -1,64 +1,61 @@
...
+    override var payload: String? =
+        JSONObject(
+            mapOf(
+                "data" to mapOf(
+                    "type" to "event",
+                    "attributes" to mapOf(
+                        "metric" to mapOf(
+                            "name" to event.name,
+                        ),
+                        "profile" to JSONObject(customerProperties.setAnonymousId().toMap()),
```
Comment (ndurell) [inline sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:48]: Do you need these sub json objects? Maybe can just do it at the top level?

File: sdk/build.gradle (hunk 2 above)
Comment (jordan-griffin) [reply sdk/build.gradle:65]: Very important

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt [hunk 4]
```diff
@@ -1,64 +1,61 @@
...
 
-    /**
-     * Generates a Unix timestamp to store as the [timestamp] on this object
-     */
-    internal fun generateUnixTimestamp() {
-        timestamp = System.currentTimeMillis() / 1000L
+    override fun appendHeaders(connection: HttpURLConnection) {
+        connection.setRequestProperty("Content-Type", "application/json")
+        connection.setRequestProperty("Accept", "application/json")
+        connection.setRequestProperty("Revision", "2022-10-17")
```
Comment (jordan-griffin) [inline sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: Should this revision number by a constant somewhere?

**Review by jordan-griffin — APPROVED** [review]
All looks good, no big issues sticking out to me!

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt (hunk 3 above)
Comment (kennyklaviyo) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:48]: maybee...can revisit this later

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt (hunk 4 above)
Comment (kennyklaviyo) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: possibly but small change can be address after the APIs are more solid

File: sdk/build.gradle (hunk 2 above)
Comment (kennyklaviyo) [reply sdk/build.gradle:65]: idk what values are possible, copied this from ktlint example

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt (hunk 4 above)
Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: Yah maybe not a bad idea but I doubt it'll be referenced anywhere else.

File: sdk/build.gradle (hunk 2 above)
Comment (ndurell) [reply sdk/build.gradle:65]: I'm being silly sorry.

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt (hunk 4 above)
Comment (jordan-griffin) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: Totally fair ya

**Review by ndurell — APPROVED** [review]
I would prefer us not to use old apis but let's tackle it later.

File: sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt (hunk 4 above)
Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: Actually if you aren't using v3 you don't need these at all.

Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: But probably harmless.

Comment (kennyklaviyo) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: track is using v3

Comment (kennyklaviyo) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: identify is still on v1

Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt:59]: Nice!

**Pull request merged**